### DISCUSSION TITLE
Is there any way to style the label that is outside textfield?

### DISCUSSION THREAD

**Discussion by Olgagr** [taiga-family/taiga-ui#290] [General]
I'm trying to add some custom styling to the label that is outside of the field.

I tried to do is with data-appearance

In css:
```
tui-wrapper[data-appearance='label-styling'] {
  text-transform: uppercase;
  color: var(--color-gray-2);
}
```

In html
```
<label tuiLabel label="Email" appearance="label-styling">
        <tui-input
          formControlName="email"
          tuiTextfieldSize="m"
          [tuiTextfieldLabelOutside]="true"
        ></tui-input>
</label>
```

It doesn't work. I also tried to use a provider:

```
{
      provide: TUI_TEXTFIELD_LABEL_OUTSIDE,
      useValue: 'label-styling',
}
```

It doesn't work as well :/ Can somebody give me a hint? Or should I do it with css and forget about taiga-ui way of component customisation...

**Comment by waterplea** [comment]
Label doesn't use wrapper inside and the token you tried to provide sets true/false for label outside (disabled flying label inside the input). You can style label with plain CSS:
https://stackblitz.com/edit/taiga-label-styling?file=src%2Fapp%2Fapp.component.css
Also `label` input supports `PolymorpheusContent` so you can have templates or dynamic components there for particularly complex layout.
